### DISCUSSION TITLE
Apply custom theme to mdx

### DISCUSSION THREAD

**Discussion by nrakochy** [storybookjs/storybook#17190] [Documentation]
I am having a bit of trouble applying a custom theme to mdx. The generated `Docs` tab works as expected on the components with `Stories` csf format, but `mdx` only is not applying the theme. I have attempted multiple variations of the [sb documentation](https://github.com/storybookjs/storybook/blob/master/addons/docs/docs/recipes.md#pure-mdx-stories). What might I be missing? Appreciate any help.

Output 

Here you can see the text is the wrong color, code source is off, etc.

<img width="291" alt="Screen Shot 2022-01-10 at 11 50 16 AM" src="https://user-images.githubusercontent.com/3740177/148817431-6914c6d7-ef15-4444-8cbf-f6ce6b53ffe7.png">

Expected Output (which applies even on the `Show Code` bit)

<img width="1069" alt="Screen Shot 2022-01-10 at 11 50 30 AM" src="https://user-images.githubusercontent.com/3740177/148817324-9347fceb-09a0-44d5-9eb0-b1f9237c5d51.png">

Here's an example `.mdx` which does not take the theme. Tried with both regular `md` inside of `mdx` as well as `Source` component

```js

import { ArgsTable, Meta, Story } from "@storybook/addon-docs";
import { Source } from "@storybook/addon-docs";
import { Prism as SyntaxHighlighter } from "react-syntax-highlighter";

import dedent from "ts-dedent";

## How to create new Storybook entries?

Storybook creation has been simplified. 

```js
const argTypes = {};
...Implementation details...

\```

## Helpful FAQ here

Detailed information

<Source
    language="tsx"
    code={dedent`type CustomProps = {
    // Remove before rendering
    $someExampleType: boolean
    // Send to the DOM
    active: boolean;
};`}
/>

```

Here's the usage in `preview.js` that works as expected for `.stories.tsx`

```js
import { dark as theme } from "./theme";

export const decorators = [];
export const parameters = {
    docs: {
        theme
    },
    actions: { argTypesRegex: "^on.*" }
};

```

**Comment by SeanCox** [comment]
If you are also using MUI I found #25134 helpful.
